### ISSUE TITLE
error: ‘nullptr’ was not declared in this scope

### ISSUE THREAD

**Issue by ouronghuang** [KhronosGroup/KTX-Software#628]
## Linux Version:

```
$ cat /etc/redhat-release
CentOS Linux release 7.9.2009 (Core)
```

## KTX Software version:

[KTX-Software-4.0.0-Source-code.zip](https://github.com/KhronosGroup/KTX-Software/archive/refs/tags/v4.0.0.zip)

## Install:

1. install cmake3

```
$ yum install -y cmake3.x86_64
$ cmake3 --version
cmake3 version 3.17.5

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

2. install KTX Software

```
$ unzip KTX-Software-4.0.0.zip
$ cd KTX-Software-4.0.0
$ cmake3 . -B build
-- Found Bash: /usr/bin/bash  
CMake Warning at cmake/version.cmake:151 (message):
  Error retrieving version from GIT tag.  Falling back to 0.0.0-noversion
Call Stack (most recent call first):
  CMakeLists.txt:10 (include)


-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Perl: /usr/bin/perl (found version "5.16.3") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /root/KTX-Software-4.0.0/build
$ cmake3 --build build
Scanning dependencies of target objUtil
[  0%] Building CXX object CMakeFiles/objUtil.dir/utils/argparser.cpp.o
In file included from /root/KTX-Software-4.0.0/utils/argparser.cpp:24:0:
/root/KTX-Software-4.0.0/utils/argparser.h:56:33: error: ‘nullptr’ was not declared in this scope
                int* longindex = nullptr);
                                 ^
/root/KTX-Software-4.0.0/utils/argparser.cpp: In member function ‘int argparser::getopt(std::string*, const argparser::option*, int*)’:
/root/KTX-Software-4.0.0/utils/argparser.cpp:75:29: error: ‘nullptr’ was not declared in this scope
         while (opt->name != nullptr) {
                             ^
/root/KTX-Software-4.0.0/utils/argparser.cpp:94:29: error: ‘nullptr’ was not declared in this scope
     } else if (shortopts != nullptr && arg.compare(0, 1, _T("-")) == 0) {
                             ^
gmake[2]: *** [CMakeFiles/objUtil.dir/utils/argparser.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/objUtil.dir/all] Error 2
gmake: *** [all] Error 2
[root@localhost KTX-Software-4.0.0]# yum install cmake3.x86_64^C
[root@localhost KTX-Software-4.0.0]# cmake3 --version
cmake3 version 3.17.5

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

## If install with rpm

```
$ rpm -ivh KTX-Software-4.0.0-Linux.rpm
error: Failed dependencies:
        libktx.so.4()(64bit) is needed by ktx-software-4.0.0-1.x86_64
        libm.so.6(GLIBC_2.27)(64bit) is needed by ktx-software-4.0.0-1.x86_64
        libstdc++.so.6(CXXABI_1.3.9)(64bit) is needed by ktx-software-4.0.0-1.x86_64
        libstdc++.so.6(GLIBCXX_3.4.20)(64bit) is needed by ktx-software-4.0.0-1.x86_64
        libstdc++.so.6(GLIBCXX_3.4.21)(64bit) is needed by ktx-software-4.0.0-1.x86_64
        libstdc++.so.6(GLIBCXX_3.4.26)(64bit) is needed by ktx-software-4.0.0-1.x86_64
```

How do I solve this problem?

**Comment by MarkCallow** [comment]
You need to use a compiler that supports at least c++11 for KTX-Software. `nullptr` was introduced in c++09.

**Comment by MarkCallow** [comment]
I am just about to release 4.1.0. I am only waiting for a new certificate for signing the Windows code. So I am no longer providing support for 4.0.0. Please try top of trunk `master` which is what will be released as 4.1.0 once I have the certificate.

All the compilers I previously used supported `nullptr` without explicitly specifying a c++ version. However 4.1.0 needs some c++11 features so the `--std` option is set in the necessary places now. The reason for the warnings you are seeing is because `add_compile_options` adds options for both c & C(++) sources.

I do not know the reason for the errors you are seeing. They code they are happening in is imported from another project via `git submodule` and that particular file has to do with SIMD support. Try turning off the `BASISU_SUPPORT_SSE` option. But first, try top of trunk.
